documentation/projects/proposals/relicensing/20241028-implementation_plan_relicensing.md

```diff
@@ -0,0 +1,301 @@
+- intent to relicense
+
+  This should be communicated once this project proposal and implementation plan
+  are approved and merged.
+
+  This notification should clearly state
+
+  - the new GNU licenses that we are planning to switch to for the various code
+    blocks
+  - that the change only applies to future commits after the license change
+  - that code before the license change can continue to be used under the same
+    terms
+
+  This also gives us time to see the community's reception to the changes.
+
+  I do not anticipate any negative reactions because, unlike the license changes
+  at Elasticseach, Redis and others, we are not making the project closed-source
+  but rather making it more open-source
+  ([in a sense](https://drewdevault.com/2024/04/19/2024-04-19-copyleft-is-not-restrictive.html))
+  by adding stronger copyleft protections.
+
+- new licenses
+
+  This should be communicated out once the license change commit is merged.
+
+  This notification should clearly state
+
+  - the new GNU licenses that we are now using the for the various code blocks
+  - that users should ensure their compliance with the terms of the new licenses
+  - that they should not use post-relicensing versions if they do not comply
+
+## Alternatives
+
+Prior to the clarification about the relicensing approach, I had considered
+dual-licensing as a viable option. All code contributed up to a specific commit
+would be licensed under the current MIT license, and all code contributed after
+that point would be licensed under the various GNU licenses we have applied to
+the various code blocks.
+
+In such cases each `LICENSE` file in the repository would have two licenses, the
+original MIT and the new GNU, with a section at the top explaining which license
+is applicable to which part of the code, based on whether it was added/modified
+before/after the cut-off date.
+
+## Design
+
+<!-- Note any design requirements for this plan. -->
+
+The communications mentioned above can use some visuals. Such visuals would be
+completely optional and subject to the availability and bandwidth of our
+designer, Francisco.
+
+## Blockers
+
+<!-- What hard blockers exist that prevent further work on this project? -->
+
+This project has no blockers but it also doesn't have any urgency to it. We
+should ideally be careful and not rush into any decisions.
+
+## Prior art
+
+<!-- Include links to documents and resources that you used when coming up with your solution. Credit people who have contributed to the solution that you wish to acknowledge. -->
+
+There is precedent for a number of web-based products switching to the GNU AGPL
+license:
+
+- [Joplin](https://joplinapp.org/news/20221221-agpl/)
+- [Plausible](https://plausible.io/blog/open-source-licenses)
+- [Grafana](https://grafana.com/blog/2021/04/20/grafana-loki-tempo-relicensing-to-agplv3/)
+- [MinIO](https://blog.min.io/from-open-source-to-free-and-open-source-minio-is-now-fully-licensed-under-gnu-agplv3/)
```

Comment:
Interesting to know that these closely used projects also underwent a similar license change!

Reply:
Yeah, it was quite interesting to see that a lot of the tools we use on the regular also transitioned to AGPL in the same way as we are planning to do. Their write-ups about the process are also very interesting to read.